### PR TITLE
Add configs for ucs-exporter

### PR DESCRIPTION
**Pull request by mithunmgore**
Added config files for alerts, templates.

Signed-off-by: Mithun Gore <mithun.gore@sap.com>

### CODE REVIEW

File: prometheus-exporters/ucs-exporter/Chart.yaml [hunk 1]
```diff
@@ -0,0 +1,7 @@
+apiVersion: v1
+name: ucs-exporter
+version: 0.1.0
+description: Exporter for UCSM/UCSC Server
+maintainers:
+  - name: Tommy Sauer (D066607)
+  - name: Mithun Gore (I355987)
```
Comment (viennaa) [inline prometheus-exporters/ucs-exporter/Chart.yaml:7]: please remove user ids

File: prometheus-exporters/ucs-exporter/Chart.yaml [hunk 2]
```diff
@@ -0,0 +1,7 @@
+apiVersion: v1
+name: ucs-exporter
+version: 0.1.0
```
Comment (viennaa) [inline prometheus-exporters/ucs-exporter/Chart.yaml:3]: let's start with 0.0.1

File: prometheus-exporters/ucs-exporter/alerts/service.alerts [hunk 3]
```diff
@@ -0,0 +1,14 @@
+groups:
+- name: license.alerts
+  rules:
+  - alert: UcsServerPortLicenseExpired
+    expr: >
+      ucs_server_license{port_name=~".+"} == 0
+    for: 5m
```
Comment (viennaa) [inline prometheus-exporters/ucs-exporter/alerts/service.alerts:7]: 15m please

File: prometheus-exporters/ucs-exporter/alerts/service.alerts [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+    labels:
+      severity: info
+      context: "ucs-exporter"
+      meta: "ucs server {{ $labels.server }} port {{ $labels.port_name }} license is expired."
+    annotations:
+      description: "ucs server {{ $labels.server }} port {{ $labels.port_name }} license is expired."
+      summary: "ucs server {{ $labels.server }} port {{ $labels.port_name }} license is expired."
```
Comment (viennaa) [inline prometheus-exporters/ucs-exporter/alerts/service.alerts:14]: Can you add a hint what needs to be done or is not working? I don't  see a playbook linked yet. Something like:
`license is expired, server ran out of support` (I am making something up here)

File: prometheus-exporters/ucs-exporter/templates/ucs-exporter-deployment.yaml [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+           ports:
+             - name: metrics
+               containerPort: {{ required ".Values.ucs.port_number missing" .Values.ucs.port_number }}
+           command:
+             - ./ucs-exporter.py
+           args:
+             - -u
+             - {{ required ".Values.ucs.user missing" .Values.ucs.user }}
+             - -p
+             - {{ required ".Values.ucs.master_password missing" .Values.ucs.master_password }}
```
Comment (viennaa) [inline prometheus-exporters/ucs-exporter/templates/ucs-exporter-deployment.yaml:37]: please change to `.Values.global.master_password`

File: prometheus-exporters/ucs-exporter/values.yaml [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+   image:
+      name: ucs-exporter
+      tag: DEFINED-IN-REGION-SECRETS
+   port_number: 9876
+   alerts:
+      enabled: false
+      prometheus: DEFINED-IN-REGION-SECRETS
+   user: DEFINED-IN-REGION-SECRETS
+   master_password: DEFINED-IN-REGION-SECRETS
+   inventory: "inventory"
```
Comment (viennaa) [inline prometheus-exporters/ucs-exporter/values.yaml:12]: what is this?

File: prometheus-exporters/ucs-exporter/values.yaml [hunk 7]
```diff
@@ -0,0 +1,21 @@
...
+   debug: "0"
+   limits:
+      memory: 100Mi
+      cpu: 450m
```
Comment (viennaa) [inline prometheus-exporters/ucs-exporter/values.yaml:16]: lets start with 250m here

File: prometheus-exporters/ucs-exporter/templates/ucs-exporter-deployment.yaml (hunk 5 above)
Comment (mithunmgore) [reply prometheus-exporters/ucs-exporter/templates/ucs-exporter-deployment.yaml]: Done

File: prometheus-exporters/ucs-exporter/values.yaml (hunk 6 above)
Comment (mithunmgore) [reply prometheus-exporters/ucs-exporter/values.yaml:12]: In program we need to get the inventory file name with -i flag, so I provided it here, not sure if its correct way. pls suggest.

File: prometheus-exporters/ucs-exporter/values.yaml (hunk 7 above)
Comment (mithunmgore) [reply prometheus-exporters/ucs-exporter/values.yaml]: Done

File: prometheus-exporters/ucs-exporter/alerts/service.alerts (hunk 4 above)
Comment (mithunmgore) [reply prometheus-exporters/ucs-exporter/alerts/service.alerts]: I checked with Kapil and got to know that, It doesnt affect the vpod traffic, but to be compliant we need to add the new license. I have edited the message, pls check.

File: prometheus-exporters/ucs-exporter/alerts/service.alerts (hunk 3 above)
Comment (mithunmgore) [reply prometheus-exporters/ucs-exporter/alerts/service.alerts]: Done

File: prometheus-exporters/ucs-exporter/Chart.yaml (hunk 2 above)
Comment (mithunmgore) [reply prometheus-exporters/ucs-exporter/Chart.yaml]: Done

File: prometheus-exporters/ucs-exporter/Chart.yaml (hunk 1 above)
Comment (mithunmgore) [reply prometheus-exporters/ucs-exporter/Chart.yaml]: Done

File: prometheus-exporters/ucs-exporter/alerts/service.alerts [hunk 8]
```diff
@@ -0,0 +1,14 @@
...
+    expr: >
+      ucs_server_license{port_name=~".+"} == 0
+    for: 15m
+    labels:
+      severity: info
+      context: "ucs-exporter"
+      meta: "ucs server {{ $labels.server }} port {{ $labels.port_name }} license is expired."
+    annotations:
+      description: "ucs server {{ $labels.server }} port {{ $labels.port_name }} license is expired."
+      summary: "ucs server {{ $labels.server }} port {{ $labels.port_name }} license is expired. Prod traffic won't be affected but to be compliant, apply new license."
```
Comment (viennaa) [inline prometheus-exporters/ucs-exporter/alerts/service.alerts:14]: What is prod traffic? Do you mean vpod traffic?

File: prometheus-exporters/ucs-exporter/templates/ucs-exporter-deployment.yaml [hunk 9]
```diff
@@ -0,0 +1,65 @@
...
+             - -c
+             - {{ .Values.ucs.config }}
+             {{- end }}
+           env:
+           - name: PORT
+             value: {{ required ".Values.ucs.port_number missing" .Values.ucs.port_number | quote }}
+           - name: DEBUG
+             value: {{ .Values.ucs.debug | quote }}
+           - name: INVENTORY
+             value: ucs-inventory
```
Comment (viennaa) [inline prometheus-exporters/ucs-exporter/templates/ucs-exporter-deployment.yaml:50]: map the values.yaml inventory var here

File: prometheus-exporters/ucs-exporter/values.yaml (hunk 6 above)
Comment (viennaa) [reply prometheus-exporters/ucs-exporter/values.yaml:12]: This need to be the path then, where the inventory file sits. Please contact me to get this sorted.

**Comment by viennaa** [comment]
@mithunmgore no updates?

**Comment by mithunmgore** [comment]
@viennaa Sorry for late update, was working on netbox to get inventory.

Here in last commit, I have removed the "inventory" var/file and added config file to get all the inventory.
Config file will be present in the top most directory so added path as only 'config'. please check.
 I will create PR for ucs-exporter with config changes soon.  Thanks

File: prometheus-exporters/ucs-exporter/values.yaml [hunk 10]
```diff
@@ -0,0 +1,21 @@
...
+   image:
+      name: ucs-exporter
+      tag: DEFINED-IN-REGION-SECRETS
+   port_number: 9876
+   alerts:
+      enabled: false
+      prometheus: DEFINED-IN-REGION-SECRETS
+   user: DEFINED-IN-REGION-SECRETS
+   master_password: DEFINED-IN-REGION-SECRETS
+   config: config
```
Comment (viennaa) [inline prometheus-exporters/ucs-exporter/values.yaml:12]: still the same problem as before, config does not map to anything here. it needs to be an absolute path in the container like /config/config.yaml

How are we feeding this in? Where is the data coming from?

**Review by viennaa — CHANGES_REQUESTED** [review]

**Comment by viennaa** [comment]
Please also add a service file to be queried from prometheus

File: prometheus-exporters/ucs-exporter/values.yaml (hunk 10 above)
Comment (viennaa) [reply prometheus-exporters/ucs-exporter/values.yaml:12]: Or is this coming from netbox now?

Comment (mithunmgore) [reply prometheus-exporters/ucs-exporter/values.yaml]: I have added the path to config. its "/ucs-exporter/config". Currently we are creating this config file manually, and I have added the "ucs_hostname_format" var which will get filled at run time with Netbox.

**Review by viennaa — APPROVED** [review]

File: prometheus-exporters/ucs-exporter/templates/ucs-config.yaml [hunk 11]
```diff
@@ -0,0 +1,14 @@
...
+kind: ConfigMap
+metadata:
+  name: ucs-config
+data:
+  config.yaml: |
+    netbox:
+        netbox_url: "https://netbox.global.cloud.sap"
+        netbox_token: "97bc508ab27df514c963435ae7097672efdaab2a"
+        ucs_hostname_format: "ucscluster-{}.cc.{}.cloud.sap"
+        regions: ["qa-de-1"]
```
Comment (viennaa) [inline prometheus-exporters/ucs-exporter/templates/ucs-config.yaml:11]: this needs to be changed to `$values.ucs.netbox_url` etc
make sure in `values.yaml` you set it to `DEFINED-IN-REGION-SECRETS` and then add it to the region secrets.

**Review by viennaa — CHANGES_REQUESTED** [review]
please change region mapping and values.yaml as stated

File: prometheus-exporters/ucs-exporter/templates/ucs-config.yaml (hunk 11 above)
Comment (mithunmgore) [reply prometheus-exporters/ucs-exporter/templates/ucs-config.yaml]: I have updated, please check if the syntax is right.

File: prometheus-exporters/ucs-exporter/values.yaml [hunk 12]
```diff
@@ -0,0 +1,26 @@
...
+   enabled: false
+   image:
+      name: ucs-exporter
+      tag: DEFINED-IN-REGION-SECRETS
+   port_number: 9876
+   alerts:
+      enabled: false
+      prometheus: DEFINED-IN-REGION-SECRETS
+   user: DEFINED-IN-REGION-SECRETS
+   master_password: DEFINED-IN-REGION-SECRETS
```
Comment (viennaa) [inline prometheus-exporters/ucs-exporter/values.yaml:11]: since your mapping is to .Values.global.master_password, this is never used and can go.

**Review by viennaa — APPROVED** [review]
let's give it a go :).

**Pull request merged**